src/lib/client/halolib.ts

```diff
@@ -0,0 +1,34 @@
+import { getCounterMessage } from "babyjubjub-ecdsa";
+
+export type RawLocationSignature = {
+  signaturePublicKey: string;
+  signatureMessage: string;
+  signature: string;
+};
+
+export const getHaLoArgs = (
+  params: URLSearchParams
+): RawLocationSignature | undefined => {
+  const pkN = params.get("pkN");
+  const rnd = params.get("rnd");
+  const rndsig = params.get("rndsig");
+
+  if (!pkN || !rnd || !rndsig) {
+    return undefined;
+  }
+
+  const strippedPkN = pkN.substring(4);
+  // Messages using Arx cards are prepended with a string before hashing,
+  // the following constructs the preimage of the hash used for the signature
+  const msgNonce = parseInt(rnd?.substring(0, 8), 16);
+  const msgRand = rnd.substring(8);
+  const signatureMessage = getCounterMessage(msgNonce, msgRand).toString(
```

Comment:
Unfortunately this is not going to work; it needs to remain in Buffer form to be correctly parsed when hashed. Ran into some bugs when using this in babyjubjub-ecdsa repo to make everything be strings, which is why the inputs to the getECDSAMessageHash is both Buffer and string lol.

Reply:
I've fixed this so that `getECDSAMessageHash` now only works with hex strings. That way, we don't have to keep track of different encoding formats - all messages for signing are encoded as hex strings.